src/tasks/task_list.hpp

```diff
@@ -89,6 +164,52 @@ class TaskList {
                          [=]() mutable -> TaskStatus { return (obj->*func)(args...); });
   }
 
+  template <class F, class... Args>
+  TaskID AddIterativeTask(const TaskType &type, const std::string &label,
+                          TaskID const &dep, F &&func, Args &&... args) {
+    if (max_iterations_.count(label) == 0) {
+      max_iterations_[label] = std::numeric_limits<unsigned int>::max();
+      count_[label] = 0;
+    }
+    if (throw_with_max_iters_.count(label) == 0) {
+      throw_with_max_iters_[label] = false;
+    }
+    if (warn_with_max_iters_.count(label) == 0) {
+      warn_with_max_iters_[label] = true;
+    }
+    TaskID id(tasks_added_ + 1);
+    task_list_.push_back(Task(
+        id, dep,
+        [=, func = std::forward<F>(func)]() mutable -> TaskStatus {
+          return func(args...);
```

Comment:
```suggestion
          return func(std::forward<Args>(args)...);
```
to maintain perfect forwarding

Reply:
fixed this and all the others.